src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/reverse_shape_and_type_infer.hpp"
+
+#include <memory>
+#include <ngraph/graph_util.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/validation_util.hpp>
+#include <openvino/opsets/opset10.hpp>
+
+#include "itt.hpp"
+#include "transformations/utils/utils.hpp"
+
+using namespace ov::opset10;
+
+namespace {
+bool inherit_output_shape(std::shared_ptr<ov::Node> node, std::vector<size_t> input_idxs) {
+    auto is_changed = false;
+    auto output_shape = node->get_output_partial_shape(0);
+
+    for (auto idx : input_idxs) {
+        if (node->get_input_partial_shape(idx).rank().is_dynamic()) {
+            OPENVINO_SUPPRESS_DEPRECATED_START
+            node->get_input_tensor(idx).set_partial_shape(output_shape);
+            OPENVINO_SUPPRESS_DEPRECATED_END
```

Comment:
@ilyachur, please pay attention to that we use publicly deprecated methods for reverse infer functionality. It is required for our purposes.

Reply:
@ilyachur, let us consider possibility to un-deprecate `set_partial_shape()` method